### DISCUSSION TITLE
Integrate gravity sync for high availability pihole setup

### DISCUSSION THREAD

**Discussion by MoJo2600** [MoJo2600/pihole-kubernetes#96] [Ideas]
Hi all,

I saw this [video on youtube](https://www.youtube.com/watch?v=IFVYe3riDRA) about [gravity sync](https://github.com/vmstan/gravity-sync) to sync between different pihole instances. Do you guys think this would be possible and could this be done? (CronJob, Sidecar).

Background is, that pihole does currently not allow to use an external database or to share the database between instances. This makes it hard create a HA setup with kubernetes.

Best Christian

**Comment by jaapio** [comment]
I would like to have this feature to make my setup more stable when there is some outage on one of the nodes.

**Comment by darkcloud784** [comment]
> Hi all,
> 
> I saw this [video on youtube](https://www.youtube.com/watch?v=IFVYe3riDRA) about [gravity sync](https://github.com/vmstan/gravity-sync) to sync between different pihole instances. Do you guys think this would be possible and could this be done? (CronJob, Sidecar).
> 
> Background is, that pihole does currently not allow to use an external database or to share the database between instances. This makes it hard create a HA setup with kubernetes.
> 
> Best Christian

Currently gravity-sync doesn't support k8's. It does support docker but k8's is switching to containerd which is separate from docker, atleast to my knowledge.
